### DISCUSSION TITLE
stdVBProject

### DISCUSSION THREAD

**Discussion by Almesi** [sancarn/stdVBA#113] [Ideas]
Hey there, ive got a class stdVBProject.
Its purpose is the handling of the Visual Basic Project, with its main goal being inclusion/import of components, creation of those components, adding or deleting code and all that with the possibility to just include it at runtime.
Are you interested in it?

**Comment by sancarn** [comment]
@Almesi Not sure, I have done something similar already here:  [VBProjectHack](https://github.com/sancarn/VBProjectHack) and here: [stdTimer](https://github.com/sancarn/stdVBA/blob/master/src/stdTimer.cls), not really sure outside of unlocking the project what there would be to gain from a wrapper. Could you perhaps elaborate?

**Reply by Almesi** [reply]
So for me i had several reasons why i build it:
First: My Companies IT Security doesnt allow me to save VBProjects with  .dll files in my code (Private Declare XXX), but they expect me to still use them to achieve the companies goals.
Second: I needed to create a system, which does not need any manual updating to the newest source code. Everything should be done automatically. But how to do this, if 100s of people are using it?
After that i created stdVBProject.
With it you can temporarily import a module with those privately declared dll files, which get removed when the workbook tries to save itself.
You can also include/replace/update a whole folder with source code to your project, so that it always stays on the newest version.
One might want to add references to the project without having to search it manually by just clicking a button.

**Reply by Almesi** [reply]
Another use case in my case is also rebuilding the project. I use another Text editor than VBE for easier text editing, but manually updating your workbook gets tedious after about 15 minutes, so i implemented a button, which reloads all components with the newest code.
I am familiar with VSCode´s xvba, but again, my company doesnt allow it.

**Reply by sancarn** [reply]
So my question really is should this be several addins or are there enough unique use cases and/or is this so difficult as to warrant a library? Take `stdArray` for instance - thousands of use cases for arrays many outside of just tracking lists of stuff, also many methods which are non-trivial to implement, it is a real building block. I think there is definitely an argument for an `xlam` which can be called from your main codebase, which imports code on demand. Importing code into the VBE just feels quite niche to me, and already not too difficult with the existing VBE API - happy to be proven otherwise though.

> My Companies IT Security doesnt allow me to save VBProjects with .dll files in my code (Private Declare XXX), but they expect me to still use them to achieve the companies goals.

P.S. ouch, this is atrocious! :/
